api/dataset.go

```diff
@@ -467,11 +541,45 @@ func (api *DatasetAPI) getDimensionOptions(w http.ResponseWriter, r *http.Reques
 	log.Debug("get dimension options", log.Data{"dataset_id": datasetID, "edition": editionID, "version": versionID, "dimension": dimension})
 }
 
-func handleErrorType(err error, w http.ResponseWriter) {
-	if err == errs.DatasetNotFound || err == errs.EditionNotFound || err == errs.VersionNotFound || err == errs.DimensionNodeNotFound || err == errs.InstanceNotFound {
-		http.Error(w, err.Error(), http.StatusNotFound)
-	} else {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
+func mapResults(results []models.DatasetUpdate) []*models.Dataset {
+	items := []*models.Dataset{}
+	for _, item := range results {
+		if item.Current == nil {
+			continue
+		}
+		item.Current.ID = item.ID
+
+		items = append(items, item.Current)
+	}
+	return items
+}
+
+func handleErrorType(docType string, err error, w http.ResponseWriter) {
+	switch docType {
```

Comment:
Could you log out the error here so we know if something has gone wrong from the server side